flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2424,11 +2438,55 @@ struct GlobalOpConversion : public FIROpConversion<fir::GlobalOp> {
     auto g = rewriter.create<mlir::LLVM::GlobalOp>(
         loc, tyAttr, isConst, linkage, global.sym_name(), initAttr);
     auto &gr = g.getInitializerRegion();
+    if (global.getType().isa<fir::SequenceType>() &&
+        !global.getType().dyn_cast<fir::SequenceType>().hasUnknownShape() &&
```

Comment:
Is it possible to create a global with unknown rank? That seems like it might be a hard error?

Reply:
It might be possible if we use the llvm.vec type instead of llvm.array.

Reply:
Not sure I follow. These lines are testing `!fir.array` types.

Reply:
OK.

What I was asking is whether it is possible to even declare an array in Fortran with unknown rank and process lifetime. (I suspect the answer is no.)